buffer/src/main/java/io/netty/buffer/api/internal/ResourceSupport.java

```diff
@@ -199,6 +201,14 @@ public boolean isAccessible() {
      */
     protected abstract Owned<T> prepareSend();
 
+    /**
+     * Called when this resource needs to be considered inaccessible.
+     * This is called at the correct points when the resource is being closed or sent,
+     * and can be used to set further traps for accesses that makes accessibility checks cheap.
+     */
+    protected void makeInaccessible() {
```

Comment:
nit: Should we note that the user is not expected to call this manually ?

Reply:
It's not entirely illegal, I would say, since sub-classes could be up to something tricky, but there's no use case at the moment so I've added a mention.